### PR TITLE
Initial NGINX Proxy Manager based load balancing solution

### PR DESCRIPTION
**Pull request by danielgrant**
This PR introduces BFI's IIIF load balancing application. It is built on top of NGINX Proxy Manager, and contains accompanying Docker Compose manifests for both local development and production deployment, plus a root certificate authority to allow the underlying NGINX instance to recognise backend services served using SSL / TLS certificates signed by that authority.

### CODE REVIEW

**Review by stephenmcconnachie — DISMISSED** [review]
LGTM, thank you

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: build
+on: [push]
+jobs:
+  build_and_push:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Extract branch name
+        run: echo "normalized_branch_name=$(stripped=${GITHUB_REF#refs/heads/} && echo ${stripped/\//-})" >> $GITHUB_ENV
```
Comment (garyttierney) [inline .github/workflows/build-and-push.yml:8]: What does GITHUB_ENV evaluate to here?

Comment (danielgrant) [reply .github/workflows/build-and-push.yml:8]: It's a built in: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable - anything you append to `$GITHUB_ENV` is then available as an environment variable in further steps.

Comment (garyttierney) [reply .github/workflows/build-and-push.yml:8]: Ah, interesting, I didn't know about that. I've been saving things as outputs to handle this in other workflows

File: deploy/etc/systemd/system/iiif-load-balancer.service [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# /etc/systemd/system/iiif-load-balancer.service
+[Unit]
+Description=IIIF Load Balancer Service
+After=docker.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+StandardError=null
+StandardOutput=null
```
Comment (garyttierney) [inline deploy/etc/systemd/system/iiif-load-balancer.service]: Does this mean we don't get logs in `journalctl`?

**Review by garyttierney — DISMISSED** [review]
One comment, otherwise LGTM.

**Pull request merged**